internal/server/middleware/authorization/authorization.go

```diff
@@ -0,0 +1,103 @@
+package authorization
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/kanopy-platform/cdnvalidator/internal/jwt"
+	log "github.com/sirupsen/logrus"
+)
+
+type ClaimsKey string
+
+const (
+	ContextBoundaryKey ClaimsKey = "claims"
+)
+
+type Entitler interface {
+	Entitled(req *http.Request, claims []string) bool
+}
+
+type Authorizer interface {
+	Authz(next http.Handler) http.Handler
+}
+
+type Middleware struct {
+	entitlementManager Entitler
+	authCookieName     string
+	authHeaderEnabled  bool
+}
```

Comment:
It's unclear why this is designed like this. Middleware is typically either a simple function that decorates an `http.Handler` ([as per our style guide](https://github.com/10gen/infrastructure-handbook/blob/master/docs/code/style/docs/go.md#server-example)), or possibly implements [Handler](https://pkg.go.dev/net/http#Handler) itself by defining a `ServeHTTP([ResponseWriter](https://pkg.go.dev/net/http#ResponseWriter), *[Request](https://pkg.go.dev/net/http#Request))` method.

Also, `Authorizer` is being defined and implemented in the same package, which is typically frowned upon, and seems like maybe instead `Middleware` just needs to implement `http.Handler`?

Can you please add some more detailed explanations as to what these interfaces are for and why things are designed this why?

Reply:
The authorizer interface was essentially dead code.    Removed that.

The main reason this was removed from the server code is to encapsulate the complex authorization logic and offer clear interfaces for other internal packages.

That being said, I did push up new commits that only expose a middleware handler function.  I think that is better given that all the members of the `middleware` struct are internal.